index.d.ts

```diff
@@ -1090,22 +1090,23 @@ export interface MenuOptionsGroupProps<T> {
   options: Array<{ value: T, label: string }>
 }
 
-export class Menu extends React.PureComponent<MenuProps> {
+declare const MenuItem: React.FC<MenuItemProps>
+declare const MenuDivider: React.FC<{}>
+declare const MenuGroup: React.FC<MenuGroupProps>
+declare const MenuOption: React.FC<MenuOptionProps>
+declare class MenuOptionsGroup<T> extends React.FC<MenuOptionsGroupProps<T>> {}
+
+export declare class Menu extends React.FC<MenuProps> {
   // @ts-ignore
-  public static Item = class MenuItem extends React.PureComponent<MenuItemProps> {
-  }
```

Comment:
Is there a better way to declare additional properties / members on a function, and have them be respected by JSX?

Reply:
Haven't yet given it a spin -- the TS compiler didn't complain (hovering over `Menu` shows that it has the right types), but lemme give it a spin